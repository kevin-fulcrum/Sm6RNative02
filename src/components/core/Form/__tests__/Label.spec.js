import React from 'react';
import {Text, View} from 'react-native';
import {shallow} from 'enzyme';
import Label from '../Label';

const initialProps = {
  label: 'Genre',
  labelStyle: {color: '#212121', fontSize: 14},
  subLabel: 'Male',
  subLabelStyle: {
    marginTop: 5,
  },
};

const wrapper = shallow(<Label {...initialProps} />);
describe('Label component', () => {
  it('should render', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('should label element in component', () => {
    expect(wrapper.find(View).length).toBe(1);
    expect(wrapper.find(Text).length).toBe(2);
  });
  it('should match props', () => {
    expect(wrapper.props.label).not.toBeNull();
    expect(wrapper.props.labelStyle).not.toBeNull();
    expect(wrapper.props.subLabel).not.toBeNull();
    expect(wrapper.props.subLabelStyle).not.toBeNull();
  });

  it('should match snapshot', () => {
    expect(wrapper.find(Label)).toMatchSnapshot();
  });
});
