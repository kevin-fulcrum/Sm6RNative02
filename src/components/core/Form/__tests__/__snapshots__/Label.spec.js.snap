// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label component should match snapshot 1`] = `null`;
